jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/CaseServiceImpl.java

```diff
@@ -164,9 +164,13 @@ public String startCase(String deploymentId, String caseDefinitionId, CaseFileIn
             ((CaseFileInstanceImpl)caseFile).setCaseId(caseId);
             logger.debug("CaseFile {} was given, associating it with case {}", caseFile, caseId);
         }
-        
-        ((CaseFileInstanceImpl)caseFile).assignOwner(newUser(identityProvider.getName()));
-        
+
+        //If owner is provided in the case file use that, otherwise default to current logged in user.
+        boolean hasOwner = ((CaseFileInstanceImpl) caseFile).getAssignments().stream().anyMatch(role -> role.getRoleName().equals(AuthorizationManager.OWNER_ROLE));
```

Comment:
This is only checking if there is a owner already set in case file when starting the case instance. I guess this is only relevant in here whereas the assignOwner is used every time there is an assignment.